src/Supportive/OutputFormatter/MermaidJSOutputFormatter.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Qossmic\Deptrac\Supportive\OutputFormatter;
+
+use Qossmic\Deptrac\Contract\OutputFormatter\OutputFormatterInput;
+use Qossmic\Deptrac\Contract\OutputFormatter\OutputFormatterInterface;
+use Qossmic\Deptrac\Contract\OutputFormatter\OutputInterface;
+use Qossmic\Deptrac\Contract\Result\OutputResult;
+use Qossmic\Deptrac\Supportive\OutputFormatter\Configuration\FormatterConfiguration;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+/**
+ * @internal
+ */
+class MermaidJSOutputFormatter implements OutputFormatterInterface
+{
+    /** @var array{direction: string, groups: array<string, string[]>} */
+    private array $config;
+
+    public function __construct(FormatterConfiguration $config)
+    {
+        /** @var array{direction: string, groups: array<string, string[]>}  $extractedConfig */
+        $extractedConfig = $config->getConfigFor('mermaidjs');
+        $this->config = $extractedConfig;
+    }
+
+    public static function getName(): string
+    {
+        return 'mermaidjs';
+    }
+
+    public function finish(
+        OutputResult $result,
+        OutputInterface $output,
+        OutputFormatterInput $outputFormatterInput
+    ): void {
+        $graph = $this->parseResults($result);
+        $violations = $result->violations();
+
+        if (null !== $outputFormatterInput->outputPath) {
+            $output = new BufferedOutput();
+        }
+
+        $output->writeln('flowchart '.$this->config['direction'].';');
+
+        if ([] !== $this->config['groups']) {
```

Comment:
These if-statements you use before the foreach loops, they don't do anything, right. For empty arrays the foreach loop will just skip the iteration.
But if checking the arrays are not empty is more performant than try to loop over them, ignore this comment.

Reply:
refactored